gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/organizations/environments/domains/UsersResource.java

```diff
@@ -148,6 +145,46 @@ public void create(
                 .subscribe(response::resume, response::resume);
     }
 
+
+    @POST
+    @Path("/bulk")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Operation(
+            operationId = "bulkCreateUser",
+            summary = "Create multiple users on the specified security domain",
+            description = "User must have the DOMAIN_USER[CREATE] permission on the specified domain " +
+                    "or DOMAIN_USER[CREATE] permission on the specified environment " +
+                    "or DOMAIN_USER[CREATE] permission on the specified organization")
+
+    @ApiResponse(responseCode = "200", description = "Some users got created, inspect each result for details",
+            content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = BulkResponse.class)))
+    @ApiResponse(responseCode = "201", description = "All users successfully created",
+            content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = BulkResponse.class)))
+    @ApiResponse(responseCode = "500", description = "Internal server error")
+    public void bulkCreate(
+            @PathParam("organizationId") String organizationId,
+            @PathParam("environmentId") String environmentId,
+            @PathParam("domain") String domainId,
+            @Parameter(name = "user", required = true)
+            @Valid @NotNull final BulkRequest<NewUser> newUsers,
```

Comment:
question: should we really use a NewUser ? I have some doubt as IMO the update user using bulk should also be done on this bulk endpoint using a POST method.

Reply:
I've asked about it either on slack or on daily, and the answer was that there will be _separate_ endpoints for different operations (create/update/delete). If not, we will need to discuss another structure